test/integration/offline.js

```diff
@@ -53,66 +53,56 @@ describe('offline', function () {
-    } finally {
-      (doc === null).should.be.true
-    }
+  it('can\'t replicate without adding the correct doctype.', async function () {
+    cozy.client.offline.createDatabase(DOCTYPE, {adapter: 'memory'})
+    const wrongDoctype = 'another.doctype'
+    return cozy.client.offline.replicateFromCozy(wrongDoctype).should.be.rejectedWith({ message: `You should add this doctype: ${wrongDoctype} to offline.` })
+  })
 
-    await new Promise(resolve => setTimeout(resolve, 5 * 1000))
+  it('can\'t replicate without creating the database in offline.', async function () {
+    const someDoctype = 'some.doctype'
+    return cozy.client.offline.replicateFromCozy(someDoctype).should.be.rejectedWith({ message: `You should add this doctype: ${someDoctype} to offline.` })
+  })
 
-    doc = null
-    err = null
+  it('can replicate created object in local database', async function () {
+    // create a database
+    const db = cozy.client.offline.createDatabase(DOCTYPE, { adapter: 'memory' })
+    // create a doc
+    const sampleDoc = { data: 'some Data' }
+    const remoteDoc = await cozy.client.data.create(DOCTYPE, sampleDoc)
+    // check the db to look for the new doc
+    db.get(remoteDoc._id).should.be.rejectedWith({ message: 'missing' })
+    // replicate the database
+    await cozy.client.offline.replicateFromCozy(DOCTYPE)
+    // doc should exist
+    return db.get(remoteDoc._id).should.be.fulfilledWith({ _id: remoteDoc._id, _rev: remoteDoc._rev, data: remoteDoc.data })
+  })
 
-    try {
-      doc = await db.get(docId)
-    } catch (e) {
-      err = e
-    } finally {
-      (err === null).should.be.true
-      doc.should.be.type('object')
-      doc._id.should.equal(docId)
-      doc.test.should.equal(smallDoc.test)
-    }
-    count.should.be.exactly(1)
-  }).timeout(7 * 1000)
+  it('can sync and stop sync a database', async function () {
+    // create a database
+    const db = cozy.client.offline.createDatabase(DOCTYPE, { adapter: 'memory' })
+    // create a doc
+    const sampleDoc = { data: 'some Data' }
+    const remoteDoc = await cozy.client.data.create(DOCTYPE, sampleDoc)
+    // check the db to look for the new doc
+    db.get(remoteDoc._id).should.be.rejectedWith({ message: 'missing' })
+    // activate synchronisation x ms
+    cozy.client.offline.startSync(DOCTYPE, 0.1)
+    // after a certain amount of time, doc should exist
+    await sleep(300)
+    cozy.client.offline.stopAllSync()
+    // create another doc after sync
+    const anotherDoc = { data: 'some other Data' }
+    const anotherRemoteDoc = await cozy.client.data.create(DOCTYPE, anotherDoc)
+    const promises = []
+    promises.push(db.get(remoteDoc._id).should.be.fulfilledWith({ _id: remoteDoc._id, _rev: remoteDoc._rev, data: remoteDoc.data }))
+    promises.push(db.get(anotherRemoteDoc._id).should.be.rejectedWith({ message: 'missing', status: 404 }))
+    cozy.client.offline.destroyDatabase(DOCTYPE)
+    return Promise.all(promises)
+  })
```

Comment:
As this is a asynchronous code and as it could have side effects (`startSync` and `stopSync`): we may add a timeout to the `it()` function to let mocha fail after a certain amount of time.